### PR TITLE
Add multizone limitations regarding dynamic provisioning

### PR DESCRIPTION
**Pull request by msau42**
Documenting known issues when using dynamic provisioning in a multizone cluster.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/4706)

### CODE REVIEW

**Comment by zacharysarah** [comment]
@msau42 :wave: Please be sure to check the box that allows edits from maintainers.

Allow edits from maintainers checkbox

File: docs/admin/multiple-zones.md [hunk 1]
```diff
@@ -73,6 +73,18 @@ available and can tolerate the loss of a zone, the control plane is
 located in a single zone.  Users that want a highly available control
 plane should follow the [high availability](/docs/admin/high-availability) instructions.
 
+* StatefulSet volume zone spreading when using dynamic provisioning is currently not compatible with
+pod affinity or anti-affinity policies.
+
+* StatefulSet volume zone spreading does not work if the StatefulSet name
```
Comment (kow3ns) [inline docs/admin/multiple-zones.md]: There was some work done to correct this. I would go with something like "If the name of the StatefulSet contains dashes ("-"), volume zone spreading may not provide a uniform distribution of storage across zones"

Comment (msau42) [inline docs/admin/multiple-zones.md]: Fixed

Comment (kow3ns) [inline docs/admin/multiple-zones.md]: /lgtm

**Pull request merged**